File: backend/FlexGCCLLC.WorkRequestTracker.Api/Features/WorkRequests/Models/WorkRequest.cs
namespace FlexGCCLLC.WorkRequestTracker.Api.Features.WorkRequests.Models;

public sealed class WorkRequest
{
    public int Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string ClientName { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public WorkRequestPriority Priority { get; set; }

    public WorkRequestStatus Status { get; set; }

    public DateTime DueDate { get; set; }

    public DateTime CreatedDate { get; set; }

    public DateTime UpdatedDate { get; set; }

    public List<WorkRequestNote> Notes { get; } = [];
}

File: frontend/src/types/workRequest.ts
export type WorkRequestPriority = 'Low' | 'Medium' | 'High'

export type WorkRequestStatus = 'New' | 'InProgress' | 'Blocked' | 'Completed'

export interface WorkRequestNote {
  id: number
  workRequestId: number
  noteText: string
  createdDate: string
}

export interface WorkRequest {
  id: number
  title: string
  clientName: string
  description: string
  priority: WorkRequestPriority
  status: WorkRequestStatus
  dueDate: string
  createdDate: string
  updatedDate: string
  notes: WorkRequestNote[]
}

export interface CreateWorkRequestInput {
  title: string
  clientName: string
  description: string
  priority: WorkRequestPriority
  status: WorkRequestStatus
  dueDate: string
}
